docs/src/app/(public)/(content)/react/handbook/forms/page.mdx

```diff
@@ -0,0 +1,523 @@
+      <Field.Label>Username</Field.Label>
+      <Field.Description>
+        May appear where you contribute or are mentioned. You can remove it at any time.
+      </Field.Description>
+      <Field.Control
+        placeholder="e.g. alice132"
+        value={value}
+        onBlur={onBlur}
+        onValueChange={onChange}
+        ref={ref}
+      />
+      <Field.Error match={!!error}>
+        {error?.message}
+      </Field.Error>
+    </Field.Root>
+  )}
+/>
+```
+
+For React Hook Form to focus invalid fields when performing validation, you must ensure that any wrapping components forward `ref`s to the underlying Base UI component. You can typically accomplish this using the `inputRef` prop, or directly as the `ref` for components that render an input element like `<NumberField.Input>`.
+
+### Field validation
+
+Specify `rules` on the `<Controller>` in the same format as [`register`](https://react-hook-form.com/docs/useform/register) options, and use the `match` prop to delegate control of the error rendering:
+
+```tsx {5-15, 33-35} title="Defining validation rules and displaying errors"
+import { Controller } from "react-hook-form"
+import { Field } from '@base-ui-components/react/field';
+
+<Controller
+  name="username"
+  control={control}
+  rules={{
+    required: 'This is a required field',
+    minLength: { value: 2, message: 'Too short' },
+    validate: (value) => {
+      if (/* custom logic */) {
+        return 'Invalid'
+      }
+      return null;
+    },
+  }}
+  render={({
+    field: { name, ref, value, onBlur, onChange },
+    fieldState: { invalid, isTouched, isDirty, error },
+  }) => (
+    <Field.Root name={name} invalid={invalid} touched={isTouched} dirty={isDirty}>
+      <Field.Label>Username</Field.Label>
+      <Field.Description>
+        May appear where you contribute or are mentioned. You can remove it at any time.
+      </Field.Description>
+      <Field.Control
+        placeholder="e.g. alice132"
+        value={value}
+        onBlur={onBlur}
+        onValueChange={onChange}
+        ref={ref}
+      />
+      <Field.Error match={!!error}>
+        {error?.message}
+      </Field.Error>
+    </Field.Root>
+  )}
+/>
+```
+
+### Submitting data
+
+Wrap your submit handler function with `handleSubmit` to receive the form values as a JS object for further handling:
+
+```tsx title="Form submission with React Hook Form"
+import { useForm } from 'react-hook-form';
+import { Form } from '@base-ui-components/react/form';
+
+interface FormValues {
+  username: string;
+  email: string;
+}
+
+const { handleSubmit } = useForm<FormValues>();
+
+async function submitForm(data: FormValues) {
+  // transform the object and/or submit it to a server
+  await fetch(/* ... */);
+}
+
+<Form onSubmit={handleSubmit(submitForm)} />;
+```
+
+## TanStack Form
+
+Base UI also supports [TanStack Form](https://tanstack.com/form/v1/docs/overview), which uses a function-based API to orchestrate validations.
```

Comment:
WDYT about rephrasing this paragraph to be similar to the `rhf`?
Now it seems that we have some "special sauce" to support this library, when it's only about integration with our components, like with `rhf`. 🤔